### PR TITLE
Optimization reporting, simplified

### PR DESCRIPTION
**Pull request by AlexandreEichenberger**
In the previous code, the instrumentation at runtime was generating garbage node names, which at times were not ASCI chars. This present PR fixes this by saving at compile time the length of the strings to be printed by instrumentation and thus making sure that exactly that number of chars is printed.

Second is that runtime stats were printed on stdout, and that is not friendly with the pipe redirecting that is happening in our python scripts.

I thus added a `ONNX_MLIR_INSTRUMENT_FILE` environment variable that enables instrumentation to write into file.
I also renamed all instrumentation env var to use the standard `ONNX_MLIR` prefix.

Using this PR, it is now pretty easy to gather combined compile-time and runtime data and analyze at once. In the `RunONNXModel*` scripts, use env var `ONNX_MLIR_INSTRUMENT_FILE` to save the runtime measurements, and `--log-to-file compile.log` option to save compile-time measurements. Then just feed the two files to `make-report.py`. 

For example:

```
> ONNX_MLIR_INSTRUMENT_FILE=run.log RunONNXModelZoo.py -c "-O3 -mcpu=z16 -maccel=NNPA \
 -opt-report=Simd --profile-ir=ZHigh" -m roberta-base-11 --log-to-file compile.log
...
> roberta-base-11 % make-report.py -c compile.log -r run.log -u us
Statistic legend:
  op-name: num, average time (us), cumulative time (us), percent of total 

Statistics start, all ops, ordered_by time, tot_time 30743.0000000
  zhigh.MatMul, 97, 293.2989691, 28450.0000000, 92.5%
  onnx.Add-simd, 87, 3.6206897, 315.0000000, 1.0%
  zhigh.Stick, 137, 1.5182482, 208.0000000, 0.7%
  zhigh.Mul, 26, 7.6153846, 198.0000000, 0.6%
  zhigh.Softmax, 12, 15.2500000, 183.0000000, 0.6%
  onnx.Mul-simd, 49, 3.0816327, 151.0000000, 0.5%
  zhigh.Unstick, 136, 1.0588235, 144.0000000, 0.5%
  zhigh.Add, 26, 5.3461538, 139.0000000, 0.5%
  onnx.Erf-simd, 12, 10.3333333, 124.0000000, 0.4%
  onnx.Div-simd, 49, 2.4285714, 119.0000000, 0.4%
  onnx.Dim, 218, 0.5229358, 114.0000000, 0.4%
  onnx.Transpose, 48, 2.3125000, 111.0000000, 0.4%
  onnx.Concat, 146, 0.6027397, 88.0000000, 0.3%
  onnx.Reshape, 144, 0.4930556, 71.0000000, 0.2%
  onnx.Add, 12, 5.5833333, 67.0000000, 0.2%
  onnx.ReduceMeanV13-simd, 50, 0.9600000, 48.0000000, 0.2%
  onnx.Constant, 93, 0.4516129, 42.0000000, 0.1%
  zhigh.StickifiedConstant, 73, 0.5273973, 38.5000000, 0.1%
  onnx.Sub-simd, 26, 0.7307692, 19.0000000, 0.1%
  onnx.Sqrt-simd, 25, 0.6400000, 16.0000000, 0.1%
  onnx.Gather, 4, 2.7500000, 11.0000000, 0.0%
  onnx.Max-simd, 12, 0.7500000, 9.0000000, 0.0%
  zhigh.Tanh, 1, 7.0000000, 7.0000000, 0.0%
  onnx.CumSum, 1, 4.0000000, 4.0000000, 0.0%
  onnx.ConstantOfShape, 2, 1.0000000, 2.0000000, 0.0%
  onnx.Unsqueeze, 2, 1.0000000, 2.0000000, 0.0%
  onnx.Equal, 1, 1.0000000, 1.0000000, 0.0%
  onnx.Mul, 1, 1.0000000, 1.0000000, 0.0%
  onnx.Not, 1, 1.0000000, 1.0000000, 0.0%
  onnx.Cast, 2, 0.0000000, 0.0000000, 0.0%
Statistics end, all ops, ordered_by time, tot_time 30743.0000000
```
 
Where the op name distinguishes if SIMD was applied or not, and where the time is the cumulative time for that op. Further details can be gathered by using the `--level` option.

### CODE REVIEW

**Review by tungld — APPROVED** [review]
LGTM.

Could we add at the end a percentage on the running time of each op? which helps quickly know which op dominates the total running time.

```
  zhigh.MatMul, 97, 15903.0000000 (87.04%)
  zhigh.Mul, 28, 323.0000000 (1.77%)
  zhigh.Stick, 139, 302.5000000 (1.66%)
  onnx.Add-simd, 87, 271.0000000 (1.48%)
```

**Comment by AlexandreEichenberger** [comment]
> Could we add at the end a percentage on the running time of each op? which helps quickly know which op dominates the total running time.

will do

File: src/Conversion/ONNXToKrnl/Math/Elementwise.cpp [hunk 1]
```diff
@@ -2094,8 +2094,12 @@ struct ONNXElementwiseBinaryOpLowering
               shapeHelper, create, op, outputMemRefType,
               collapsedInnermostLoops, estimatedSimdLoopTripCount);
       if (uVL > 0) {
-        onnxToKrnlSimdReport(op, /*successful*/ true, uVL,
-            estimatedSimdLoopTripCount, "binary with manageable broadcast");
+        if (collapsedInnermostLoops == outputRank)
```
Comment (sorenlassen) [inline src/Conversion/ONNXToKrnl/Math/Elementwise.cpp:2097]: @AlexandreEichenberger fyi, clang 14.0.3 on Mac warns:
```
Elementwise.cpp:2097:37: warning: comparison of integers of different signs: 'int64_t' (aka 'long long') and 'uint64_t' (aka 'unsigned long long') [-Wsign-compare]
        if (collapsedInnermostLoops == outputRank)
```

Comment (AlexandreEichenberger) [reply src/Conversion/ONNXToKrnl/Math/Elementwise.cpp:2097]: @sorenlassen thanks, saw it too and fixed it in a subsequent PR. Thanks for the info

**Pull request merged**